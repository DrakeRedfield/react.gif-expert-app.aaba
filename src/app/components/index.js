export * from './gif-expert-app/gif-expert-app';
export * from './gif-card/gif-card';
export * from './gif-grid/gif-grid';
export * from './add-category/add-category';