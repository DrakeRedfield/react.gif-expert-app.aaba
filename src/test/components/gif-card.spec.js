import { render, screen } from "@testing-library/react"
import { GifCard } from "../../app/components"

describe('Gif Card Testing', () => {
  const title = 'Uncharted';
  const url = 'https://mock.url.test/testing-url';

  test('should match with snapshot', () => {
    const { container } = render(<GifCard title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
  test('should render with title and url', () => {
    render(<GifCard title={title} url={url} />);
    const { alt, src } = screen.getByRole('img');

    expect(alt).toBe(title);
    expect(src).toBe(url);
  });
  test('should render title', () => {
    render(<GifCard title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
})