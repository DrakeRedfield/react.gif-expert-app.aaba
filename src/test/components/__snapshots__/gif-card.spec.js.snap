// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif Card Testing should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Uncharted"
      src="https://mock.url.test/testing-url"
    />
    <p>
      Uncharted
    </p>
  </div>
</div>
`;
